src/main/scala/djinni/CppMarshal.scala

```diff
@@ -122,7 +122,7 @@ class CppMarshal(spec: Spec) extends Marshal(spec) {
             List(ImportRef("<memory>"))
           }
           spec.cppNnHeader match {
-            case Some(nnHdr) => ImportRef(nnHdr) :: base
+            case Some(nnHdr) => ImportRef(q(nnHdr.replace("\"", ""))) :: base
```

Comment:
I changed this because I couldn't find a way to generate exactly `#include "nn.hpp"`, given [this line of code](https://github.com/cross-language-cpp/djinni-generator/blob/main/src%2Fit%2Fscala%2Fdjinni%2FIntegrationTest.scala#L146)
```
     if (useNNHeader) {
>      cmd += " --cpp-nn-header nn.hpp"
       cmd += " --cpp-nn-type dropbox::oxygen::nn_shared_ptr"
       cmd += " --cpp-nn-check-expression NN_CHECK_ASSERT"
     }
```

I remembered when I tried to change that line to `cmd += """ --cpp-nn-header "nn.hpp""""` or many alternatives (e.g., `\"`), it would generate something like `#include ""nn.hpp""`. And I remembered when I managed to make it work in the tests, it doesn't work when I use the compiled djinni.bat to generate my real code; but it could be something to do with the batch script. I need to check again the details of the issue tomorrow.

So the work around is, we don't depend on whether user passes also double quotes in `--cpp-nn-header` or not, but we first remove all the double quotes, then add double quotes back. Then it works in both tests and my code.

Probably you have better ideas what could be the issue of generating `#include ""nn.hpp""`, then we can find more elegant way to solve the issue.

Reply:
That's a good catch! Seems there is a general problem 

There is a  `def q(s: String) = '"' + s + '"'` in the generator

Very often `ImportRef` is used like this : `ImportRef(q(d.name + ".hpp"))`  but not always ... and this might be the problem,  I guess that is not the most easy to read code in the code base, and it makes also not super clear how it should be used, but that's how it is and we got it 🙈  

maybe that line should be changed to 
```scala
spec.cppNnHeader match {
  case Some(nnHdr) => ImportRef(q(nnHdr)) :: base
```
and then 
`cmd += " --cpp-nn-header nn.hpp"` 
works 
and if we 🙏 , and have luck, nothing breaks 😁  

Can you please give that a try, and if it breaks something, tell us what, so we can unify ... (but it should work, the line is only handling (should) the cmd line parameter, just wonder how it worked so far, or was it untested ... 🤔

Reply:
I feel ignored, but that's OK

logically speaking, one of both must have the right quoting
either `ImportRef(q(nnHdr))` or `ImportRef(nnHdr)`

if this is not the case, then there is something special in the input (the input has been q'ed in one case but not an other) , and if there is something special in the input, it might be interesting to know where this comes from, since then this might also happen on other places. Not? 
And if, we might need to change the q function (optimally also rename but this is a different story) to ensure something is `"` as it should, and not to much or to less

Since we do no needed that special handling anywhere else, I would be really interested in the reasons

Reply:
@a4z eventually, I realized several issues:

1. The script I used to generate the code looks like: `--cpp-nn-header "nn.hpp"` which is with double quote. We needed to add double quotes because the generator doesn't add them automatically:
    ```
    spec.cppNnHeader match {
      case Some(nnHdr) => ImportRef(nnHdr) :: base
    ```
2. Current tests do not cover the case of generating `#include "nn.hpp"`. In the `main` branch, although a test set `useNNHeader` to true (see [this line](https://github.com/cross-language-cpp/djinni-generator/blob/5bb1846ac4db44aa53a4720623f1465214522a90/src/it/scala/djinni/GeneratorIntegrationTest.scala#L333), it doesn't generate `#include "nn.hpp"`. But if we set `objc = true`, `java = true`, and `cppCLI = true`, we will see the issue:
    ```c++
    // jni-headers/dependent_interface.hpp
    #include nn.hpp

    // objcpp-headers/ITDependentInterface+Private.h
    #include nn.hpp

    // cppcli/DependentInterface.hpp
    // It doesn't generate #include nn.hpp in the current `main` branch.
    ```
3. In my branch, this issue happens because I added a test generating C++/CLI code with `useNNHeader = true`.

I will change the generator for all C++, JNI, Objective-C++, and C++/CLI in my branch, but let's deal with the tests separately:

## C++ (`CppMarshal.scala`)
```
spec.cppNnHeader match {
    case Some(nnHdr) => ImportRef(q(nnHdr)) :: base
...
```

## JNI (`JNIGenerator.scala`)
```
spec.cppNnHeader match {
    case Some(nnHdr) => jniHpp.add("#include " + q(nnHdr))
...
```

## Objective-C++ (`ObjcppGenerator.scala`)
```
spec.cppNnHeader match {
    case Some(nnHdr) => refs.privHeader.add("#include " + q(nnHdr))
...
```

## C++/CLI (`CppCliGenerator.scala`)
```
spec.cppNnHeader match {
    case Some(nnHdr) =>
        refs.hpp.add("#include " + q(nnHdr))
...
```

Then, if someone who wrote a batch / bash script to generate the code with `--cpp-nn-header "path/to/nn.hpp"` option before, would need to remove double quotes in their script.

Reply:
@a4z I apologize that you feel ignored 😓, when I was still typing the comment, your most recent reply was not shown yet... Hope my above explanation helps to understand the issue a bit better and we can find an agreement how to deal with the case

Reply:
No worries, didn't mean it very serious. 
I am sorry if my way of expressing me is clumsy and made you feel you have to apologize, now I have to do so. Sorry!

Thanks you so much for the fix, that looks much better and consistent now! Awesome!!

Reply:
Haha, let me grab my broken heart pieces and glue them again (🤲💔 🪄❤️‍🩹), no worries 😄.